src/execution_plan/ops/op_aggregate.c

```diff
@@ -197,26 +198,31 @@ static Record _handoff(OpAggregate *op) {
 	return r;
 }
 
-OpBase *NewAggregateOp(const ExecutionPlan *plan, AR_ExpNode **exps, bool should_cache_records) {
+OpBase *NewAggregateOp(const ExecutionPlan *plan, AR_ExpNode **projection_exps,
+					   AR_ExpNode **order_exps) {
 	OpAggregate *op = rm_malloc(sizeof(OpAggregate));
-	op->exps = exps;
 	op->group = NULL;
 	op->group_iter = NULL;
 	op->group_keys = NULL;
 	op->expression_classification = NULL;
 	op->non_aggregated_expressions = NULL;
 	op->groups = CacheGroupNew();
-	op->exp_count = array_len(exps);
+	op->should_cache_records = (order_exps != NULL);
+
+	// Migrate ORDER expressions to the projection array as appropriate.
+	CombineProjectionArrays(&projection_exps, &order_exps);
+	assert(!order_exps && "Aggregate operation does not resolve all ORDER BY expressions"); // TODO tmp
```

Comment:
Is there any AST validation which prevents this?
if not can we issue a run-time error?
Remove `// TODO tmp`

Reply:
This is here because https://github.com/RedisGraph/RedisGraph/pull/899#discussion_r374258867 still holds true - if we're happy with the ProjectOp changes, they need to be mirrored in OpAggregate. I'll implement them here on Monday and remove this assertion.

There are two syntax validations we don't implement currently that would reduce the frequency of this scenario for aggregate ops:
`Cannot use aggregation in ORDER BY if there are no aggregate expressions in the preceding RETURN`
`In a WITH/RETURN with DISTINCT or an aggregation, it is not possible to access variables declared before the WITH/RETURN`
But they wouldn't remove the need for this extra logic entirely.